### PR TITLE
Split buried/suspended in statistics view

### PR DESCRIPTION
**Pull request by jdpopkin**
## Purpose / Description
I use templates that produce a lot of siblings, and want to be able to see my current number of buried cards without actually typing in a search. Generally, I do this by using the Statistics view to see my number of "Suspended+Buried" cards at the start of the day. If I remember that number, I can just check "Suspended+Buried" again later in the day to get a sense of how many buried cards I must have. That's kind of silly, though, and ever since I noticed desktop Anki split "Suspended+Buried" into separate "Suspended" and "Buried" stats, I've wanted the Android app to do the same.

## Fixes
I haven't filed an issue to request this feature yet. Should I do that?

## Approach
This change just extends what AnkiDroid already does to draw the card-types pie chart. It makes the "suspended+buried" subquery specific to suspended cards, adds another subquery for buried cards, and updates various places where AnkiDroid reads the results of the card-types query.

## How Has This Been Tested?
I created a new deck, turned on sibling burial, created a cloze note with two deletions, and created a handful of basic cards.  I suspended one of the basic cards, marked another as easy, and manually buried a third. When I drew the cloze card, I marked it as easy.

Then I opened the Statistics tab and confirmed that the type-counts were correct and the pie chart looked reasonable.

I did this using an emulator (Pixel 2 API 30).

This is the first time I've done any sort of Android development, so I don't know which devices or versions are likely to run into problems with this type of change. I'd be happy to repeat my tests against other configurations if that would be helpful.

## Learning (optional, can help others)
I looked around at what desktop Anki does. I think [this](https://github.com/ankitects/anki/blob/41d74592f0068428992b8e7cfd27a874c6da3823/ts/src/stats/card-counts.ts#L28) is the most relevant function.

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

I'm not sure where this fits into the template, but: I didn't update the localization key-name (per [the Development Guide in the wiki](https://github.com/ankidroid/Anki-Android/wiki/Development-Guide#handling-translations)). But it seems like that might be a good idea. `"statistics_suspended_and_buried"` is really just `"statistics_suspended"` now. Same thing is true of the `"stats_suspended_and_buried"` color attribute - maybe I can change that one safely without messing up existing translations?

Also, I'm not sure I handled the new colors correctly. I just added an arbitrary shade of gray for "Buried". I didn't poke around in desktop Anki to make sure the colors match, since I assume Ankidroid's color schemes are more complicated/configurable than desktop Anki's. Let me know if I should go find the colors and make them match.

### CODE REVIEW

File: AnkiDroid/src/main/res/values/06-statistics.xml [hunk 1]
```diff
@@ -22,7 +22,8 @@
     <string name="statistics_mature">Mature</string>
     <string name="statistics_young_and_learn">Young+Learn</string>
     <string name="statistics_unlearned">Unseen</string>
-    <string name="statistics_suspended_and_buried">Suspended+Buried</string>
+    <string name="statistics_suspended_and_buried">Suspended</string>
```
Comment (mikehardy) [inline AnkiDroid/src/main/res/values/06-statistics.xml:25]: I think your mention of this in the PR was your intuition feeling it is probably better to to go ahead and change this, and I think it is one of the cases where the old key needs to go away and a fresh key needs to go in, as this is semantically different now. It's great to be sensitive to the translator workload but semantic differences should be respected

```suggestion
    <string name="statistics_suspended">Suspended</string>
```

Please note, I have not looked to see if the old key was used elsewhere! It very well might be, then I'd argue for addition of this key I'm suggesting without deleting the old key

File: AnkiDroid/src/main/res/values/attrs.xml [hunk 2]
```diff
@@ -85,6 +85,7 @@
     <attr name="stats_counts" format="color"/>
     <attr name="stats_unseen" format="color"/>
     <attr name="stats_suspended_and_buried" format="color"/>
```
Comment (mikehardy) [inline AnkiDroid/src/main/res/values/attrs.xml:87]: same thought process here as above with the translation though this doesn't ripple out to translations at least - *if* `stats_suspended_and_buried` is not used elsewhere in the combination, then I'd vote for altering the key in all cases (colors themes etc) to be like so

```suggestion
    <attr name="stats_suspended" format="color"/>
```

**Review by mikehardy — CHANGES_REQUESTED** [review]
Wow, this looks pretty good on a first pass. Thanks, and welcome!
I left some comments which are basically just echoing your thought process / doubt on the translation, but with a vote to go ahead and add a new key for suspended by itself, with a deletion of the old key pending a codebase grep to see if it is used elsewhere (at which point old key remains but new key is still a good idea)

I don't imagine this will have problems across our supported API skew, doesn't seem to be that type of code, and shouldn't matter on real devices either, so API30 emu testing should be sufficient.

Cheers

File: AnkiDroid/src/main/res/values/06-statistics.xml (hunk 1 above)
Comment (jdpopkin) [reply AnkiDroid/src/main/res/values/06-statistics.xml]: Okay, I've renamed both this key and the color-key and tested that change. Looks like things still work.

The only change that felt sketchy was in [lint-baseline.xml](https://github.com/ankidroid/Anki-Android/blob/master/AnkiDroid/lint-baseline.xml). Based on the commit history, I think that usage is safe to delete (hiding linter complaints for a problem that can't happen once that key's gone). I deleted it, but let me know if I should restore (or restore and replace the old key with the new one).

With that deletion, I think the old key is only present in the translation files. `git grep 'suspended_and_buried' -- ':!AnkiDroid/src/main/res/**/*'` doesn't find anything.

**Comment by mikehardy** [comment]
Hello! with the 2.14 release we started accepting donations with the stated goal of accepting money from users to support maintenance and development of AnkiDroid

This PR was included in 2.14, and if you'd like to submit an expense to share in a portion of our hopefully-sustainable monthly disbursement, please read here and do so in the next couple of days:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process

This is our first time handling donations and attempting to compensate anyone with these processes, so please be flexible and patient with us

Thank you for your contribution to AnkiDroid!

**Pull request merged**